django/restapi/views.py

```diff
@@ -33,15 +39,32 @@
 
 
 # /login
-@api_view(['GET'])
-def login(request):
-    return JsonResponse({})
+class LoginView(KnoxLoginView):
+    permission_classes = (permissions.AllowAny,)
+
+    def post(self, request, format=None):
+        serializer = AuthTokenSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
```

Comment:
Shouldn't the exception be caught and a 500 HTTP error code be returned?

Reply:
No, [500](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/500) is for internal server problems. Currently it returns [401 Unauthorized](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401) and some detail like
```json
{
	"detail": "Authentication credentials were not provided."
}
```
or
```json
{
	"detail": "Invalid token."
}
```